src/main/java/ch/njol/skript/expressions/ExprItemFlags.java

```diff
@@ -0,0 +1,95 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.ItemFlag;
+import org.bukkit.inventory.meta.ItemMeta;
+import ch.njol.skript.aliases.ItemType;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@Name("Item Flags")
+@Description("Returns or modifies the item flags of an item.")
+@Examples({
+	"set item flags of player's tool to hide enchants and hide attributes",
+	"add hide potion effects to item flags of player's held item",
+	"remove hide enchants from item flags of {legendary sword}"
+})
+@Since("INSERT VERSION")
+public class ExprItemFlags extends PropertyExpression<ItemType, ItemFlag> {
+	static {
+		register(ExprItemFlags.class, ItemFlag.class, "item flags", "itemtypes");
+	}
```

Comment:
I don't remember but, would you not need to override the `isSingle` as the default would be true or `getExpr().isSingle()` no?